.circleci/config.yml

```
@@ -0,0 +1,233 @@
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/project/app-starter
+  docker:
+    - image: cimg/node:14.17.0
+
+aws-credentials: &aws-credentials
+  aws-access-key-id: AWS_ACCESS_KEY
+  aws-secret-access-key: AWS_ACCESS_SECRET
+
+orbs:
+  node: circleci/node@4.5.1
+  aws-cli: circleci/aws-cli@1.3
+  aws-ecr: circleci/aws-ecr@7.0.0
+  pulumi: pulumi/pulumi@2.0.0
+  shellcheck: circleci/shellcheck@2.2.4
+  kubernetes: circleci/kubernetes@0.12.0
+
+jobs:
+  lint:
+    <<: *defaults
+    steps:
+      - checkout
+      - node/install-packages
+      - run:
+          name: Lint
+          command: npm run lint
+  setup-env-dev: &setup-env
+    <<: *defaults
+    steps:
+      - checkout
+      - run:
+          name: Setup environment variables
+          command: source ./.circleci/setup-env.sh
+      - persist_to_workspace:
+          root: .
+          paths:
+            - .env
+  setup-env-staging:
+    <<: *setup-env
+  setup-env-prod:
+    <<: *setup-env
+  build-client-dev: &build-client
+    <<: *defaults
+    steps:
+      - checkout
+      - node/install-packages
+      - run:
+          name: Build client
+          command: npm run build:client
+      - persist_to_workspace:
+          root: .
+          paths:
+            - dist/client
+  build-client-staging:
+    <<: *build-client
+  build-client-prod:
+    <<: *build-client
+  build-server-dev: &build-server
+    working_directory: ~/project/app-starter
+    executor: aws-ecr/default
+    steps:
+      - aws-cli/setup:
+          <<: *aws-credentials
+          aws-region: AWS_REGION
+      - aws-ecr/build-and-push-image:
+          <<: *aws-credentials
+          attach-workspace: true
+          create-repo: true
+          repo: "${AWS_ECR_REPO}"
+          tag: "${CIRCLE_SHA1}"
+  build-server-staging:
+    <<: *build-server
+  build-server-prod:
+    <<: *build-server
+  deploy-dev: &deploy
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run: cd ./infrastructure && npm i
+      - aws-cli/setup:
+          <<: *aws-credentials
+          aws-region: AWS_REGION
+      - kubernetes/install-kubectl
+      - pulumi/login
+      - run:
+          name: Stack init
```

Sorry for nitpicking, but please consider whether this is nicer for the eyes 😅 
```suggestion
          name: Configure stack
```